src/main/java/com/hillel/dzhurynskyi/homework/hm17/Products.java

```diff
@@ -0,0 +1,75 @@
+package com.hillel.dzhurynskyi.homework.hm17;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.hillel.dzhurynskyi.homework.hm17.TypesOfProducts.*;
+
+public class Products {
+    private List<Product> products;
+
+    public void setProducts() {
```

Comment:
Setters for establishing something. This is may confusing. In my opinion, better to have it in the `Main.java`